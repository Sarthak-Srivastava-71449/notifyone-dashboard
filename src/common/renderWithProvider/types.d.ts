/*
    Auth User Types
*/

interface IUser {
  loading: boolean;
  error: string;
  success: boolean | number;
  name: string;
  tokens: {
    accessToken: string;
  };
  user_info: {
    email_id: string;
    contact_number: string;
    email_verified: boolean;
  };
  roles: IRoles;
}

type UserAction = {
  type?: string;
  payload?: IUser;
};

type UserActionDispatchType = (args: UserAction) => UserAction | any;

interface AxiosGoogleResponse {
  auth_token: string;
  auth_token_labs: string;
  token: string;
  username: string;
}

interface UserInfoResponse {
  roles: IRoles;
  user_info: {
    email_id: string;
    contact_number: string;
    email_verified: boolean;
  };
}
