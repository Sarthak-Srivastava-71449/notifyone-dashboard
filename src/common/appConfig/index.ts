const AppConfig: any = process.env.communication;

export default AppConfig;
