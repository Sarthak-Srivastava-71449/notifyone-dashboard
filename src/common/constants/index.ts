export const COMMUNICATION_TYPE = {
  All: 'All',
  SMS: 'SMS',
  Whatsapp: 'Whatsapp',
  Transactional_Push_notification: 'Transactional Push Notification',
  Email: 'Email',
};

export const MAX_INPUT_LIMIT = 30;

export const AUTH_ROLES = 'LARA_ADMIN';

export const EVENT_FETCH_LIMIT = 50;

// Navigation
export const COMS_TEMPLATE_NAVIGATE = '/communication/template';
