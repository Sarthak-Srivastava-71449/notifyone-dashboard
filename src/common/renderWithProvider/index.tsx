export { default as withProvider } from './renderWithProvider';
export { store, withLocationProvider } from './renderWithProvider';
export {
  mockEmailEventDetails,
  mockPushNotificationEventDetails,
  mockSmsEventDetails,
  mockWhatsappEventDetails,
} from './constants';
