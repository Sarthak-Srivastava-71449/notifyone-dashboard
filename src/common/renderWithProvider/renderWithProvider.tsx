import React from 'react';
import thunk from 'redux-thunk';
import {
  applyMiddleware,
  combineReducers,
  legacy_createStore as createStore,
} from 'redux';
import { composeWithDevTools } from '@redux-devtools/extension';
import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { ToastContainer } from 'react-toastify';

import messages from '../i18n/messages';
import googleReducer from 'src/platform/reducers/auth.reducer';

import emailEventsReducer from 'src/store/reducers/emailEvents.reducer';
import pushNotificationEventsReducer from 'src/store/reducers/pushNotificationEvents.reducer';
import smsEventsReducer from 'src/store/reducers/smsEvents.reducer';
import whatsAppEventsReducer from 'src/store/reducers/whatsappEvents.reducer';

import Dashboard from 'src/view/components/Dashboard/Dashboard';
import CommunicationList from 'src/view/components/CommunicationList/CommunicationList';
import Communication from 'src/view/components/Communication/Communication';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock('antd', () => {
  const antd = jest.requireActual('antd');

  return {
    ...antd,
  };
});

jest.mock('common/src/components/Spinner', () => ({
  Spinner: () => {
    return <></>;
  },
}));

let store: any;

const createTestStore = () => {
  const store = createStore(
    combineReducers({
      emailEvents: emailEventsReducer,
      pushNotificationEvents: pushNotificationEventsReducer,
      smsEvents: smsEventsReducer,
      whatsAppEvents: whatsAppEventsReducer,
      user: googleReducer,
    }),
    composeWithDevTools(applyMiddleware(thunk))
  );
  return store;
};

const withProvider = (component: any) => {
  store = createTestStore();
  store.dispatch = jest.fn();
  const locale = 'en-us';

  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={messages[locale]}>
        <BrowserRouter>{component}</BrowserRouter>
      </IntlProvider>
    </Provider>
  );
};

const withLocationProvider = (to: string, state: any = {}) => {
  store = createTestStore();
  store.dispatch = jest.fn();
  const locale = 'en-us';

  window.history.pushState(state, '', to);

  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={messages[locale]}>
        <BrowserRouter>
          <ToastContainer />
          <Routes>
            <Route path="communication" element={<Dashboard />}>
              <Route path="templates" element={<CommunicationList />} />
              <Route path="template" element={<Communication />} />
            </Route>
          </Routes>
        </BrowserRouter>
      </IntlProvider>
    </Provider>
  );
};

export { store, withLocationProvider };
export default withProvider;
