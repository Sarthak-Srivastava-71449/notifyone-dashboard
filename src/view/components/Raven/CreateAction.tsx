import React, { Fragment, useEffect, useReducer, useState } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { Select, Input } from 'antd';
import { Button } from '../Button';
import './Raven.css';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from 'src/store';
import {
  fetchRavenMetaData,
  fetchRavenTickets,
} from 'src/store/actions/ravenRootNodeEvents';
import {
  addNewNodeAction,
  fetchRavenNodeActions,
} from 'src/store/actions/ravenActionNodeEvents';
import { useNavigate } from 'react-router-dom';
import ravenConstants from 'src/common/constants/ravenConstants';
import { toast } from 'react-toastify';
import CheckBoxFields from './CheckBoxFields';
import {
  getCreateSuccess,
  getRavenMetaInfo,
  getTicketOption,
} from 'src/store/selectors/createAction';
import { getAccessToken } from 'src/store/selectors/accessToken';

const CreateAction: React.FC = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  /*
    Redux Selectors for fetching data
  */

  const ticketPriorityObjects: Array<any> = useSelector((state: RootState) =>
    getTicketOption(state)
  );
  const ravenMetaInfo = useSelector((state: RootState) =>
    getRavenMetaInfo(state)
  );
  const accessToken = useSelector((state: RootState) => getAccessToken(state));
  const createSuccess = useSelector((state: RootState) =>
    getCreateSuccess(state)
  );

  /*
   UseEffect for loading Raven Node Actions Data
  */

  useEffect(() => {
    if (createSuccess === true) {
      dispatch(fetchRavenNodeActions(accessToken, 200, 0));
      navigate('/communication/raven', { replace: true });
    }
  }, [createSuccess]);

  const [optionalFields, setOptionalFields] = useState<Array<string>>([]);
  const [requiredFields, setRequiredFields] = useState<Array<string>>([]);

  /*
   UseEffect for Fetching Raven Meta Data and Ticket Data
 */

  useEffect(() => {
    dispatch(fetchRavenMetaData(accessToken));
    dispatch(fetchRavenTickets(accessToken));
  }, []);

  const [isPreview, setIsPreview] = useState(false);

  const editActionInputState = {
    headerName: '',
    screenText: '',
    responseText:
      '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.',
    type: '',
    ticketPriority: '',
  };

  /*
   React reducer for managing inputState
  */

  const editActionReducer = (state = editActionInputState, action: any) => {
    switch (action.type) {
      case 'INPUT_CHANGE':
        return { ...state, ...action.payload };
    }
  };
  const [editState, editDispatch] = useReducer(
    editActionReducer,
    editActionInputState
  );

  const updateInputField = (event: any, inputField: string) => {
    const payload: { [key: string]: string } = {};
    const input = event?.target?.value;
    payload[inputField] = input;
    setIsPreview(false);
    editDispatch({
      type: 'INPUT_CHANGE',
      payload: payload,
    });
  };

  const { Option } = Select;

  const intl = useIntl();

  const createNewAction = () => {
    const payload = {
      application: 'PHARMACY',
      emails: '',
      event: '',
      header: editState.headerName.trim(),
      optional_fields: optionalFields.length > 0 ? optionalFields : null,
      required_fields: requiredFields.length > 0 ? requiredFields : null,
      response: editState.responseText,
      screen_text: editState.screenText,
      ticket: editState.ticketPriority,
      type: editState.type,
    };
    const [error, data] = validateData(payload);
    if (error === false && data !== null) {
      dispatch(addNewNodeAction(accessToken, data));
    }
  };

  /*
   Function for validate data in form
  */

  const validateData = (payload: any): [boolean, null | object] => {
    let error = false;
    const submitData = { ...payload };
    if (submitData.type === 'INFO' || submitData.type === 'WEBVIEW') {
      submitData.optional_fields = null;
      submitData.required_fields = null;
      if (submitData.screen_text === '' || submitData.screen_text === null) {
        toast.error(ravenConstants.INFO_WEBVIEW_CHECK);
        error = true;
      }
    }
    if (submitData.type !== '' && submitData.response !== '') {
      if (
        submitData.type === 'FORM' ||
        submitData.type === 'INFO_WITH_WRITE_TO_US'
      ) {
        let fieldsLength = 0;
        if (submitData.optional_fields !== null) {
          fieldsLength += submitData.optional_fields.length;
        }
        if (submitData.required_fields !== null) {
          fieldsLength += submitData.required_fields.length;
        }
        if (fieldsLength === 0) {
          toast.error(ravenConstants.FIELDS_LENGTH_CHECK);
          error = true;
        }
        if (
          submitData.required_fields &&
          submitData.required_fields.indexOf('skus') > -1 &&
          fieldsLength !== 3
        ) {
          toast.error(ravenConstants.SKU_TYPE_CHECK);
          error = true;
        } else if (
          ((submitData.required_fields &&
            submitData.required_fields.indexOf('attachments') > -1) ||
            (submitData.optional_fields &&
              submitData.optional_fields.indexOf('attachments') > -1)) &&
          fieldsLength !== 3
        ) {
          toast.error(ravenConstants.ATTACHMENT_TYPE_CHECK);
          error = true;
        }
      }
    } else {
      toast.error(ravenConstants.NAME_TYPE_CHECK);
      error = true;
    }
    return [error, submitData];
  };

  /*
   Function for Optional field update
  */

  const optionalFieldUpdate = (fieldValue: string) => {
    if (optionalFields.includes(fieldValue)) {
      const newArr = [...optionalFields];
      const index = newArr.indexOf(fieldValue);
      newArr.splice(index, 1);
      setOptionalFields(newArr);
    } else {
      setOptionalFields([fieldValue, ...optionalFields]);
    }
  };

  /*
   Function for required field update
  */

  const requiredFieldUpdate = (fieldValue: string) => {
    if (requiredFields.includes(fieldValue)) {
      const newArr = [...requiredFields];
      const index = newArr.indexOf(fieldValue);
      newArr.splice(index, 1);
      setRequiredFields(newArr);
    } else {
      setRequiredFields([fieldValue, ...requiredFields]);
    }
  };

  const { TextArea } = Input;
  return (
    <Fragment>
      <div className="raven-wrapper">
        <div className=" mx-auto wallet-wrapper raven-wrap  shadow-md">
          <h2 className="text-lg raven-form-title font-semibold text-center py-1">
            <FormattedMessage id="create_action_form" />
          </h2>
          <div className=" py-2 px-3 flex-col flex">
            <div className="flex mb-3  gap-4">
              <div className="flex-1">
                <span className="text-sm font-semibold">
                  <FormattedMessage id="node_action_header" />
                  <span className="px-1">:</span>
                </span>
                <span>
                  <Input
                    className={
                      'custom-input  focus:ring-blue-500 focus:outline-none appearance-none w-full text-sm leading-6  rounded-md py-1 px-1.5  shadow-sm'
                    }
                    placeholder={intl.formatMessage({ id: 'header_name' })}
                    type="text"
                    data-testid="headerName"
                    value={editState.headerName}
                    onChange={(e) => updateInputField(e, 'headerName')}
                  />
                  <div className="no-character">
                    {editState?.headerName?.trim().length > 0 && (
                      <span className="character">
                        <span className="text-sm font-semibold">
                          <FormattedMessage id="no_of_characters" />
                        </span>{' '}
                        <span className="px-1">:</span>
                        {editState?.headerName.trim().length}
                      </span>
                    )}
                  </div>
                </span>
              </div>
            </div>
            <div className="flex mb-3 gap-4">
              <div className="input-label flex-1">
                <span className="text-sm font-semibold">
                  <FormattedMessage id="type_details" />
                  <span className="px-1">:</span>
                </span>
                <span>
                  <Select
                    placeholder={intl.formatMessage({ id: 'select_type' })}
                    className="w-full"
                    value={editState.type}
                    onChange={(value: string) => {
                      setIsPreview(false);
                      editDispatch({
                        type: 'INPUT_CHANGE',
                        payload: { type: value },
                      });
                      // Set Ticket Priority To None
                      editDispatch({
                        type: 'INPUT_CHANGE',
                        payload: { ticketPriority: '' },
                      });
                    }}
                  >
                    {ravenMetaInfo &&
                      ravenMetaInfo?.type?.map((type: any) => (
                        <Option key={type.id} value={type.type}>
                          {type.type}
                        </Option>
                      ))}
                  </Select>
                </span>
              </div>
              <div className="flex-1">
                <span className="text-sm font-semibold ">
                  <FormattedMessage id="ticket_priorty" />
                  <span className="px-1">:</span>
                </span>
                <span>
                  <Select
                    placeholder={intl.formatMessage({
                      id: 'select_ticket_priority',
                    })}
                    className="w-full"
                    value={editState.ticketPriority}
                    onChange={(value: string) => {
                      setIsPreview(false);
                      editDispatch({
                        type: 'INPUT_CHANGE',
                        payload: { ticketPriority: value },
                      });
                    }}
                  >
                    <Option key={0} value={''}>
                      None Priority
                    </Option>
                    {ticketPriorityObjects &&
                      ticketPriorityObjects?.map((obj) => (
                        <Option key={obj?.id} value={obj?.id}>
                          {obj?.status + '_' + obj?.priority}
                        </Option>
                      ))}
                  </Select>
                </span>
              </div>
            </div>
            <div className="flex mb-3 flex-col mt-3 gap-4">
              <div className="input-label flex-1">
                <span className="text-sm font-semibold">
                  <FormattedMessage id="optional_fields" />
                  <span className="px-1">:</span>
                  {ravenMetaInfo &&
                    ravenMetaInfo?.fields?.map((nodeActionFields: any) => (
                      <CheckBoxFields
                        key={nodeActionFields.id}
                        changeFunction={optionalFieldUpdate}
                        name={'optional_fields'}
                        id={nodeActionFields.id}
                        fields={nodeActionFields.fields}
                        checkedField={optionalFields}
                        disabledField={requiredFields}
                        isFieldChecked={false}
                        skusCheck={
                          nodeActionFields.fields === 'skus' ? true : false
                        }
                      />
                    ))}
                </span>
                <span></span>
              </div>
              <div className="flex-1">
                <span className="text-sm font-semibold">
                  <FormattedMessage id="required_fields" />
                  <span className="px-1">:</span>
                  {ravenMetaInfo &&
                    ravenMetaInfo?.fields?.map((nodeActionFields: any) => (
                      <CheckBoxFields
                        key={nodeActionFields.id}
                        changeFunction={requiredFieldUpdate}
                        name={'required_fields'}
                        id={nodeActionFields.id}
                        fields={nodeActionFields.fields}
                        checkedField={requiredFields}
                        disabledField={optionalFields}
                        isFieldChecked={false}
                        skusCheck={false}
                      />
                    ))}
                </span>
                <span></span>
              </div>
            </div>
            <div className="flex mb-3 flex-col gap-4">
              <div className="input-label flex-1">
                <span className="text-sm font-semibold">
                  <FormattedMessage id="screen_text" />
                  <span className="px-1">:</span>
                </span>
                <span>
                  <TextArea
                    rows={4}
                    placeholder={`<b>Thank you!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.`}
                    value={editState.screenText}
                    onChange={(e) => updateInputField(e, 'screenText')}
                  />
                  <div className="no-character">
                    {editState?.screenText?.length > 0 && (
                      <span className="character">
                        <span className="text-sm font-semibold">
                          <FormattedMessage id="no_of_characters" />
                        </span>{' '}
                        <span className="px-1">:</span>
                        {editState?.screenText?.length}
                      </span>
                    )}
                    {isPreview && editState.screenText.length > 0 && (
                      <div
                        className="screen-test"
                        dangerouslySetInnerHTML={{
                          __html: editState.screenText,
                        }}
                      />
                    )}
                  </div>
                </span>
              </div>
              <div className="flex-1">
                <span className="text-sm font-semibold">
                  <FormattedMessage id="response_text" />
                  <span className="px-1">:</span>
                </span>
                <span>
                  <TextArea
                    rows={4}
                    placeholder={`<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.`}
                    value={editState.responseText}
                    onChange={(e) => updateInputField(e, 'responseText')}
                  />
                  <div className="no-character">
                    {editState?.responseText?.length > 0 && (
                      <span className="character">
                        <span className="text-sm font-semibold">
                          <FormattedMessage id="no_of_characters" />
                        </span>{' '}
                        <span className="px-1">:</span>{' '}
                        {editState?.responseText?.length}
                      </span>
                    )}
                  </div>
                  {isPreview && editState.responseText.length > 0 && (
                    <div
                      className="screen-test"
                      dangerouslySetInnerHTML={{
                        __html: editState.responseText,
                      }}
                    />
                  )}
                </span>
              </div>
            </div>
            <div className="flex justify-center items-center">
              {!isPreview ? (
                <Button className="" onClick={() => setIsPreview(true)}>
                  <FormattedMessage id="preview" />
                </Button>
              ) : (
                <Button className="" onClick={() => createNewAction()}>
                  <FormattedMessage id="submit" />
                </Button>
              )}
            </div>
          </div>
        </div>
      </div>
    </Fragment>
  );
};
export default CreateAction;
