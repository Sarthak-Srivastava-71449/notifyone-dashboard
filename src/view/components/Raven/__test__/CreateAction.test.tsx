import React from 'react';
import { withProvider } from '../../../../common/renderWithProvider';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import CreateAction from '../CreateAction';
import userEvent from '@testing-library/user-event';
import * as redux from 'react-redux';
import { ticketPriorityObjects, accessToken } from './mock';

jest.mock('src/store/actions/ravenNodeEvents');

describe('Create Action Test-Suite', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterEach(cleanup);
  test('Match Create Node Action Snapshot', async () => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => ticketPriorityObjects)
      .mockImplementationOnce(() => accessToken)
      .mockImplementationOnce(() => false);
    const { asFragment } = render(withProvider(<CreateAction />));
    expect(asFragment()).toMatchSnapshot();
  });

  test('Showcase Mock Data', async () => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => ticketPriorityObjects)
      .mockImplementationOnce(() => accessToken)
      .mockImplementationOnce(() => false);

    render(withProvider(<CreateAction />));
  });

  test('React Internationalization - Should Work As Expected', () => {
    const component = render(withProvider(<CreateAction />));
    expect(component.container).toHaveTextContent('Create Node Action Form');
  });

  test('Header Name input should be clickable and editable and showcase error accordingly', async () => {
    const component = render(withProvider(<CreateAction />));
    const HeaderNameInput = screen.getByPlaceholderText(
      'Header Name'
    ) as HTMLInputElement;
    await userEvent.type(HeaderNameInput, 'Create Node Action');
    expect(HeaderNameInput).toHaveValue('Create Node Action');
    await userEvent.clear(HeaderNameInput);
    expect(HeaderNameInput).toHaveValue('');

    // Type Special characters and numbers to showcase errors
    await userEvent.type(HeaderNameInput, '23232');
    expect(component.container).toHaveTextContent(
      'Create Node Action FormNode Action Header:No of Characters :5Type Details:Ticket Priority:None PriorityOptional Fields:Required Fields:Screen Text:Response Text:<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.No of Characters : 122Preview'
    );
  });

  test('Create Node Action Form Screen Text Area', () => {
    render(withProvider(<CreateAction />));
    const input = screen.getByPlaceholderText(
      '<b>Thank you!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.'
    );
    fireEvent.change(input, { target: { value: '23' } });
  });

  test('Create Node Action Form Response Text Area', () => {
    render(withProvider(<CreateAction />));
    const input = screen.getByDisplayValue(
      '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.'
    );
    fireEvent.change(input, {
      target: {
        value:
          '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.',
      },
    });
  });
});
