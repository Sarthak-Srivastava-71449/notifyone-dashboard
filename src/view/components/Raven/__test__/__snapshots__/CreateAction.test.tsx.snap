// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Action Test-Suite Match Create Node Action Snapshot 1`] = `
<DocumentFragment>
  <div
    class="raven-wrapper"
  >
    <div
      class=" mx-auto wallet-wrapper raven-wrap  shadow-md"
    >
      <h2
        class="text-lg raven-form-title font-semibold text-center py-1"
      >
        Create Node Action Form
      </h2>
      <div
        class=" py-2 px-3 flex-col flex"
      >
        <div
          class="flex mb-3  gap-4"
        >
          <div
            class="flex-1"
          >
            <span
              class="text-sm font-semibold"
            >
              Node Action Header
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span>
              <input
                class="ant-input custom-input  focus:ring-blue-500 focus:outline-none appearance-none w-full text-sm leading-6  rounded-md py-1 px-1.5  shadow-sm"
                data-testid="headerName"
                placeholder="Header Name"
                type="text"
                value=""
              />
              <div
                class="no-character"
              />
            </span>
          </div>
        </div>
        <div
          class="flex mb-3 gap-4"
        >
          <div
            class="input-label flex-1"
          >
            <span
              class="text-sm font-semibold"
            >
              Type Details
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span>
              <div
                class="ant-select w-full ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title=""
                  />
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </span>
          </div>
          <div
            class="flex-1"
          >
            <span
              class="text-sm font-semibold "
            >
              Ticket Priority
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span>
              <div
                class="ant-select w-full ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="None Priority"
                  >
                    None Priority
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </span>
          </div>
        </div>
        <div
          class="flex mb-3 flex-col mt-3 gap-4"
        >
          <div
            class="input-label flex-1"
          >
            <span
              class="text-sm font-semibold"
            >
              Optional Fields
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span />
          </div>
          <div
            class="flex-1"
          >
            <span
              class="text-sm font-semibold"
            >
              Required Fields
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span />
          </div>
        </div>
        <div
          class="flex mb-3 flex-col gap-4"
        >
          <div
            class="input-label flex-1"
          >
            <span
              class="text-sm font-semibold"
            >
              Screen Text
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span>
              <textarea
                class="ant-input"
                placeholder="<b>Thank you!</b><br>We have received your message and will be in touch over your registered email id as soon as possible."
                rows="4"
              />
              <div
                class="no-character"
              />
            </span>
          </div>
          <div
            class="flex-1"
          >
            <span
              class="text-sm font-semibold"
            >
              Response Text
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <span>
              <textarea
                class="ant-input"
                placeholder="<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible."
                rows="4"
              >
                &lt;b&gt;Thank You!&lt;/b&gt;&lt;br&gt;We have received your message and will be in touch over your registered email id as soon as possible.
              </textarea>
              <div
                class="no-character"
              >
                <span
                  class="character"
                >
                  <span
                    class="text-sm font-semibold"
                  >
                    No of Characters
                  </span>
                   
                  <span
                    class="px-1"
                  >
                    :
                  </span>
                   122
                </span>
              </div>
            </span>
          </div>
        </div>
        <div
          class="flex justify-center items-center"
        >
          <button
            class="button primary-button "
            data-testid="custom-button"
          >
            Preview
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
